### PR TITLE
Remove format button should work with collapsed selection

### LINKED ISSUE #4008: Remove Format button no longer works when no text is selected
## Type of report

Bug

## Provide detailed reproduction steps (if any)

1. Load the editor version 4.14.0: https://jsfiddle.net/xda4omnr/1/
2. Select "bold" and "underlined"
3. Type: `Hello, `
4. Click "Remove Format" button
5. Type: `world!`

### Expected result

The text `world!` has normal style.

### Actual result

The text `world!` is bold and underlined.

## Other details

This feature used to work before - up to and including version 4.11.1. See https://jsfiddle.net/xda4omnr/2/ where the feature works.

The culprit which introduced the regression is most likely this commit: https://github.com/ckeditor/ckeditor4/pull/2625/commits/4a2929658c62f15d0c721e0e5c3027b7be40cf09

Note that you can still remove formatting after selecting some text, but some users use this button just to reset their styling from a particular moment on without selecting any text.

* Browser: Chrome
* OS: Mac OS
* CKEditor version: 4.14.0
* Installed CKEditor plugins: full/all

### PR DESCRIPTION
**Pull request by jacekbogdanski**
## What is the purpose of this pull request?

Bug fix

## Does your PR contain necessary tests?

All patches that change the editor code must include tests. You can always read more
on [PR testing](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_contributing_code.html#tests),
[how to set the testing environment](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_tests.html) and
[how to create tests](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_tests.html#creating-your-own-test)
in the official CKEditor documentation.

### This PR contains

- [x] Unit tests
- [x] Manual tests

## Did you follow the CKEditor 4 code style guide?

Your code should follow the guidelines from the [CKEditor 4 code style guide](https://github.com/ckeditor/ckeditor4/blob/major/dev/docs/codestyle.md) which helps keep the entire codebase consistent.

- [x] PR is consistent with the code style guide

## What is the proposed changelog entry for this pull request?

```
*[#4008](https://github.com/ckeditor/ckeditor4/issues/4008): Fixed: [Remove format](https://ckeditor.com/cke4/addon/removeformat) doesn't work with collapsed selection.
```

## What changes did you make?

I reverted changes introduced in https://github.com/ckeditor/ckeditor4/pull/2625 as they seem not necessary and adds this regression. Remove format operates only on inline elements, which is nicely documented here: https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_config.html#cfg-removeFormatTags
Also, it seems like remove format indeed only breaks inline elements:https://github.com/ckeditor/ckeditor4/blob/846b07e0d7780ff1826685ed7a9f69c169c0c6d1/plugins/removeformat/plugin.js#L63-L64

so, enlarging range only for inline elements may be a good approach. I didn't note any special issues with skipping block elements and the issue https://github.com/ckeditor/ckeditor4/issues/2451 is still covered.

## Which issues does your PR resolve?

Closes #4008

### CODE REVIEW

File: tests/plugins/removeformat/plugin.js [hunk 1]
```diff
@@ -173,5 +173,15 @@ bender.test(
 		editor.execCommand( 'removeFormat' );
 
 		assert.beautified.html( html, bender.tools.selection.getWithHtml( editor ), { customFilters: [ filter ] } );
+	},
+
+	// (#4008)
+	'test remove format with collapsed selection': function() {
+		this.editorBot.setHtmlWithSelection( '<p><strong>Hello,^</strong></p>' );
+
+		this.editor.execCommand( 'removeFormat' );
```
Comment (Comandeer) [inline tests/plugins/removeformat/plugin.js:182]: `this.editor` is used several times in the test – maybe extract it to a variable?

**Comment by jacekbogdanski** [comment]
Weird, it seems to work correctly for me (on IE11):

removeformat

Nevertheless, changing this test to always request full input should cover some edge cases :+1:
